### ISSUE TITLE
DataGrid - The editorType option does not work if defined in the formItem configuration object

### ISSUE THREAD

**Issue by bbakermmc** [DevExpress/DevExtreme#7003]
Update the MVC API to allow for the grid to edit form items, or remove it from an available options to edit.

Seems like a huge waste to make a template to do something so trivial.  Is there a better way to tell it how I want it to render?

EX:  I can configure a form item here, I would expect it to render.  But per documentation it wont, so dont expose it to me then. 
```c#
c.AddFor(f => f.isActive)
           .CellTemplate(@<text> <% if(value){ %>
                             <span class='badge badge-success'> Yes</span><% }else{ %>
                             <span class='badge badge-danger'> No</span> <%}%> </text>)
           .FormItem(fi => fi.Editor(e => e.Switch()
                                           .SwitchedOnText("Yes")
                                           .SwitchedOffText("No")));
```

**Comment by vconst** [comment]
Hi @bbakermmc,

Currently, it is not possible to define editor type in the formItem config.
We'll update our help topic accordingly.
You can change editor type as described at https://js.devexpress.com/Documentation/Guide/Widgets/DataGrid/Editing/#Customize_Editors
In your case the code may look as follows:
```
.OnEditorPreparing(@<text> function(e) {
    if(e.dataField === "isActive" && e.rowType === "data") {
        e.editorOptions.editorType = "dxSwitch";
    }
} </text>)
```
I'd like to add that we understand that this restriction is not convenient and are going to find a way to overcome it in the future.

**Comment by bbakermmc** [comment]
Yeah I know. But we didn’t pay extra for the Mvc core helpers to end up writing just as much JS.

Sent from my iPhone

On Feb 25, 2019, at 3:10 AM, Konstantin Volnyagin <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @bbakermmc<https://github.com/bbakermmc>,

Currently, it is not possible to define editor type in the formItem config.
We'll update our help topic accordingly.
You can change editor type as described at https://js.devexpress.com/Documentation/Guide/Widgets/DataGrid/Editing/#Customize_Editors
In your case the code may look as follows:

.OnEditorPreparing(@<text> function(e) {
    if(e.dataField === "isActive" && e.rowType === "data") {
        e.editorOptions.editorType = "dxSwitch";
    }
} </text>)


I'd like to add that we understand that this restriction is not convenient and are going to find a way to overcome it in the future.

**Comment by onebalaban** [comment]
We fixed this issue in v19.1. You can see the result in the [Form Editing](https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/FormEditing/Mvc/Light/) demo.

**Comment by urlimited** [comment]
> We fixed this issue in v19.1. You can see the result in the [Form Editing](https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/FormEditing/Mvc/Light/) demo.

Thanks a lot, man you have saved me lot of time
